pkg/reconciler/cache/replication/replication_controller.go

```diff
@@ -60,20 +63,28 @@ func NewController(
 	dynamicLocalClient kcpdynamic.ClusterInterface,
 	localKcpInformers kcpinformers.SharedInformerFactory,
 	globalKcpInformers kcpinformers.SharedInformerFactory,
+	localKubeInformers kcpkubernetesinformers.SharedInformerFactory,
+	globalKubeInformers kcpkubernetesinformers.SharedInformerFactory,
 ) (*controller, error) {
 	c := &controller{
-		shardName:                      shardName,
-		queue:                          workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName),
-		dynamicCacheClient:             dynamicCacheClient,
-		dynamicLocalClient:             dynamicLocalClient,
-		localAPIExportLister:           localKcpInformers.Apis().V1alpha1().APIExports().Lister(),
-		localAPIResourceSchemaLister:   localKcpInformers.Apis().V1alpha1().APIResourceSchemas().Lister(),
-		localShardLister:               localKcpInformers.Core().V1alpha1().Shards().Lister(),
-		localWorkspaceTypeLister:       localKcpInformers.Tenancy().V1alpha1().WorkspaceTypes().Lister(),
-		globalAPIExportIndexer:         globalKcpInformers.Apis().V1alpha1().APIExports().Informer().GetIndexer(),
-		globalAPIResourceSchemaIndexer: globalKcpInformers.Apis().V1alpha1().APIResourceSchemas().Informer().GetIndexer(),
-		globalShardIndexer:             globalKcpInformers.Core().V1alpha1().Shards().Informer().GetIndexer(),
-		globalWorkspaceTypeIndexer:     globalKcpInformers.Tenancy().V1alpha1().WorkspaceTypes().Informer().GetIndexer(),
+		shardName:             shardName,
+		queue:                 workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName),
+		dynamicKcpCacheClient: dynamicCacheClient,
+		dynamicKcpLocalClient: dynamicLocalClient,
+
+		localAPIExportLister:          localKcpInformers.Apis().V1alpha1().APIExports().Lister(),
+		localAPIResourceSchemaLister:  localKcpInformers.Apis().V1alpha1().APIResourceSchemas().Lister(),
+		localShardLister:              localKcpInformers.Core().V1alpha1().Shards().Lister(),
+		localWorkspaceTypeLister:      localKcpInformers.Tenancy().V1alpha1().WorkspaceTypes().Lister(),
+		localClusterRoleLister:        localKubeInformers.Rbac().V1().ClusterRoles().Lister(),
+		localClusterRoleBindingLister: localKubeInformers.Rbac().V1().ClusterRoleBindings().Lister(),
+
+		globalAPIExportIndexer:          globalKcpInformers.Apis().V1alpha1().APIExports().Informer().GetIndexer(),
+		globalAPIResourceSchemaIndexer:  globalKcpInformers.Apis().V1alpha1().APIResourceSchemas().Informer().GetIndexer(),
+		globalShardIndexer:              globalKcpInformers.Core().V1alpha1().Shards().Informer().GetIndexer(),
+		globalWorkspaceTypeIndexer:      globalKcpInformers.Tenancy().V1alpha1().WorkspaceTypes().Informer().GetIndexer(),
+		globalClusterRoleIndexer:        globalKubeInformers.Rbac().V1().ClusterRoles().Informer().GetIndexer(),
```

Comment:
why do we need to replicate cr and crb ?

Reply:
As explained in gmeet: these are required to authorize binding of APIExports, and for maximum permission policy, for now. There will be more reasons for other cross-workspace authorizations.